solver/src/main/java/org/chocosolver/solver/constraints/reification/PropXltYCReif.java

```diff
@@ -43,8 +43,8 @@ public void propagate(int evtmask) throws ContradictionException {
                     this.setPassive();
                 }
             } else {
-                vars[0].updateLowerBound(vars[1].getLB() + cste, this);
-                vars[1].updateUpperBound(vars[0].getUB() - cste, this);
+                while(vars[0].updateLowerBound(vars[1].getLB() + cste, this) |
```

Comment:
is the while still necessary after using the views?
if it is used to handle the case where x=y, would it be possible to catch it before and simplify as
if (c > 0) then b = 1
else b = 0
?

Reply:
Actually, that propagator can be safe regarding the non-uniqueness of variables by adding this line of code. 

I should have mentioned that there are 3 ways to fix issues related to multiple occurrences of a variable in a propagator:
1. ensuring that the propagator is indempotent
2. duplicating the propagator
3. using views

The first one should be favored but the the other ones are easy to setup.

Reply:
ok thanks, but regarding my second point. 
if the constraint is "b <=> x < x + c"
If c > 0, the constraint is always true so we should set b to 1 and set propagator to passive
Else we should set b to false and set propagator to passive

I believe it would make the code clearer and more efficient, no?